wallet/src/de/schildbach/wallet/ui/HeaderBalanceFragment.java

```diff
@@ -101,6 +106,7 @@ public void onAttach(final Activity activity) {
         this.activity = (AbstractBindServiceActivity) activity;
         this.application = (WalletApplication) activity.getApplication();
         this.config = application.getConfiguration();
+        hideBalance = config.getHideBalanceOnLaunch();
```

Comment:
Given this requirement:

If the setting for Hide Balance = ON then the balance should always be hidden when returning to the home screen (regardless of the last state)

hideBalance should be set also in `onResume()`

Reply:
It shouldn't because state is stored in hideBalance